### PR TITLE
Reporting API ADR

### PR DESCRIPTION
**Pull request by julioalex-rezende**
First draft for Reporting API covering Endpoints, Service and Model Domains

### CODE REVIEW

File: data-reconciliation-app/docs/adr/05-Reporting.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+//TODO Add reference to this future ADR: https://github.com/microsoft/ccf-app-samples/issues/102
+
+
+## Service
+
+### Reporting Service
+
+The reporting service will take the record(s) from KV-store and create SummaryResult object(s).
+
+- **For all data:** This service will be responsible to iterate the KV-store checking whether the member has submitted data against the key or not.
```
Comment (kusweta) [inline data-reconciliation-app/docs/adr/05-Reporting.md:83]: This is going to be an expensive operation. Is this only for the demo purpose?
Apologies, if it has already been discussed earlier

File: data-reconciliation-app/docs/adr/05-Reporting.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+{
+  "key": "<record key>",
+  "value": "<value informed by User>",
+  "group_status": "<value>",
+  "total_votes_count": "#",           // total opinions count - initially comented (DEMO CHANGE)
+  "count_of_unique_values": "#",         // count of unique values
+  "members_in_agreement": "#",              // # of members in agreement with the User value
+  "majority_minority": "<calculated value>"
+  "statistics": {
+      // extra columns for number statistics.
```
Comment (kusweta) [inline data-reconciliation-app/docs/adr/05-Reporting.md:107]: What statistics are we planning here?

File: data-reconciliation-app/docs/adr/05-Reporting.md (hunk 1 above)
Comment (mkcomer) [reply data-reconciliation-app/docs/adr/05-Reporting.md:83]: Definitely going to be expensive...we have discussed this and noted it as a conseuqence to our current repository design: https://github.com/microsoft/ccf-app-samples/blob/main/data-reconciliation-app/docs/adr/01-data-ingest.md#consequences

I added a story to the backlog to investigate our repository model and see if there is a better design to improve performance: https://github.com/orgs/microsoft/projects/542/views/1

Good observation! @kusweta

File: data-reconciliation-app/docs/adr/05-Reporting.md (hunk 2 above)
Comment (mkcomer) [reply data-reconciliation-app/docs/adr/05-Reporting.md:107]: +1 is there a need for statistics field? Appears to me that all of the [requirements](https://github.com/microsoft/ccf-app-samples/blob/main/data-reconciliation-app/docs/data-schema-data-flow.md#data-results-schema) are captured in the object already?

File: data-reconciliation-app/docs/adr/05-Reporting.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+### Summary Result Object
+
+The SummaryResult Object will be the final output object in the report.
+
+```json
+{
+  "key": "<record key>",
+  "value": "<value informed by User>",
+  "group_status": "<value>",
+  "total_votes_count": "#",           // total opinions count - initially comented (DEMO CHANGE)
```
Comment (mkcomer) [inline data-reconciliation-app/docs/adr/05-Reporting.md:102]: Agree we will want to comment this total_votes_count out and add it in for demo change

**Review by mkcomer — APPROVED** [review]
LGTM! Only had a few comments on the summary results object. Thanks for capturing this in a ADR.

**Review by ross-p-smith — APPROVED** [review]

File: data-reconciliation-app/docs/adr/05-Reporting.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+Proposed
+
+## Context
+
+We need to build an API that reports the members' reconciled dataset back to them.
+
+## Requirements:
+- A report can be requested for a single or for all records.
+- Members will receive a report only for data they have ingested.
+- A reconciliation report can be issued at anytime, by any member, and the report result will be presented based on the most recent data present in the KV-store for the entire network.
```
Comment (ross-p-smith) [inline data-reconciliation-app/docs/adr/05-Reporting.md:14]:
```suggestion
- A reconciliation report can be requested at anytime, by any member, and the report result will be presented based on the most recent data present in the KV-store for the entire network.
```

File: data-reconciliation-app/docs/adr/05-Reporting.md (hunk 2 above)
Comment (julioalex-rezende) [reply data-reconciliation-app/docs/adr/05-Reporting.md:107]: In a future iteration we may want to expand the Model to accept quantitative statistics such as mean/std/etc. Thus I captured here.

But I agree with you and by the time we decide to make this change, we design this change :)

File: data-reconciliation-app/docs/adr/05-Reporting.md (hunk 1 above)
Comment (julioalex-rezende) [reply data-reconciliation-app/docs/adr/05-Reporting.md:83]: Yep. I'm not happy with that either but that's how it will be for now :/

**Pull request merged**